### ISSUE TITLE
[need help] optimizeBlockSize: NVCC failed

### ISSUE THREAD

**Issue by yangliu2** [genn-team/genn#400]
I was just trying to run tutorial 1 for ver 4.4.0, but this error occurs after `model.build()`. 

```
<ipython-input-2-a3fa8a0d9271> in <module>
     14        "n": 0.5961207}  # prob. for K channel activation
     15 pop1 = model.add_neuron_population("Pop1", 10, "TraubMiles", p, ini)
---> 16 model.build()

/data/Downloads/genn-4.4.0/pygenn/genn_model.py in build(self, path_to_model)
    510         # Create backend
    511         backend = self._backend_module.create_backend(self._model, share_path, output_path, 
--> 512                                                       self.backend_log_level, preferences);
    513 
    514         # Generate code

RuntimeError: optimizeBlockSize: NVCC failed
```

Tested `nvcc --version` works fine. 
Not sure if it matters if I'm running in a docker container. 
Also heard genn requires cuda 10.0, but can't find it in documentation. 
Tried to get a cuda 10 docker, but instead his this error when building from source
`backend.h:15:10: fatal error: cuda.h: No such file or directory`

**Comment by neworderofjamie** [comment]
GeNN doesn't require CUDA 10 (I tested with 7.5 recently!) so maybe start without the docker. Regarding the nvcc problem, can you compile e.g. a CUDA example with NVCC and does the nvidia-smi command line tool work correctly?

**Comment by yangliu2** [comment]
Thanks for the quick reply and tips. I tried to move this outside of the docker container. I have different errors. Confirmed I can compile CUDA examples and nvidia-smi runs and shows cuda 11.2. 

Detailed error in docker:
```
nvcc fatal   : Value 'sm_86' is not defined for option 'gpu-architecture'
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-a3fa8a0d9271> in <module>
     14        "n": 0.5961207}  # prob. for K channel activation
     15 pop1 = model.add_neuron_population("Pop1", 10, "TraubMiles", p, ini)
---> 16 model.build()

/data/Downloads/genn-4.4.0/pygenn/genn_model.py in build(self, path_to_model)
    510         # Create backend
    511         backend = self._backend_module.create_backend(self._model, share_path, output_path, 
--> 512                                                       self.backend_log_level, preferences);
    513 
    514         # Generate code

RuntimeError: optimizeBlockSize: NVCC failed
```
When using anaconda outside of docker, error is:
```
gcc: fatal error: cannot execute ‘cc1plus’: execvp: No such file or directory
compilation terminated.
nvcc fatal   : Failed to preprocess host compiler properties.
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-a3fa8a0d9271> in <module>
     14        "n": 0.5961207}  # prob. for K channel activation
     15 pop1 = model.add_neuron_population("Pop1", 10, "TraubMiles", p, ini)
---> 16 model.build()

~/anaconda3/envs/genn/lib/python3.9/site-packages/pygenn/genn_model.py in build(self, path_to_model)
    509 
    510         # Create backend
--> 511         backend = self._backend_module.create_backend(self._model, share_path, output_path, 
    512                                                       self.backend_log_level, preferences);
    513 

~/anaconda3/envs/genn/lib/python3.9/site-packages/pygenn/genn_wrapper/CUDABackend.py in create_backend(model, sharePath, outputPath, backendLevel, preferences)
    133 
    134 def create_backend(model: "ModelSpecInternal const &", sharePath: "std::string const &", outputPath: "std::string const &", backendLevel: "plog::Severity", preferences: "Preferences") -> "CodeGenerator::CUDA::Backend":
--> 135     return _CUDABackend.create_backend(model, sharePath, outputPath, backendLevel, preferences)
    136 
    137 def delete_backend(backend: "CodeGenerator::CUDA::Backend *") -> "void":

RuntimeError: optimizeBlockSize: NVCC failed
```

**Comment by neworderofjamie** [comment]
Are you using a prebuilt PyGeNN wheel from our releases page or building PyGeNN from source? Also, can you try using the system python rather than anaconda? Anaconda includes a strange version of gcc which isn't compatible with CUDA (and hence GeNN)

**Comment by yangliu2** [comment]
Appreciate it! I upgraded g++ and it seems to work in Ananconda. I'm too much of a freak to run anything on system python. The clean up is just not worth it. The docker version maybe due to I have too new of a video card. It's from last year, so TF need to keep up. Apparently the Tensorflow based docker image I built this from, have a problem working with my card. I will follow up if I get docker container to work.
